### ISSUE TITLE
Cursor appearance get broken with 0.9.14

### ISSUE THREAD

**Issue by djfd** [LibVNC/x11vnc#42]
for 0.9.13 it looks like this:
![vnc-0 9 13](https://cloud.githubusercontent.com/assets/126338/24174919/38f3e418-0edf-11e7-9239-dca2242a969a.png)

and after upgrading to 0.9.14 I got:
![vnc-0 9 14](https://cloud.githubusercontent.com/assets/126338/24174935/4dfde188-0edf-11e7-9750-1a5fda2b6453.png)

nothing else was changed

env: Arch linux, Xorg, gnome, all is up to date

**Comment by apocalyptech** [comment]
Yeah, this is happening to me as well.  Other cursors get similarly munged up.  The cursors do at least change when they're supposed do, but they're all wrong in similar ways.

I've verified through the Perl module X11::Protocol::Other (specifically its `examples/xfixes-cursor-image.pl` script) that my X server's XFIXES extension is reading the cursors properly.

**Comment by bk138** [comment]
As this seems a regression, you could help by finding the commit to blame via https://git-scm.com/docs/git-bisect - thanks a lot!

**Comment by apocalyptech** [comment]
Sure, looks like I got a result with that (hadn't actually seen `git bisect` before, quite handy.  Wasn't sure how closely I would be able to pinpoint it since it turns out immediately after the "bad" commit, there's a string of eight or so commits which either don't compile or freeze up a second or two into the VNC session, but in the end it turned out to be right before that run.

Anyway, here you go:

```
a9c95879b8822dac524c5ca6d74575f41bcf6914 is the first bad commit
commit a9c95879b8822dac524c5ca6d74575f41bcf6914
Author: Christian Beier <dontmind@freeshell.org>
Date:   Sun Dec 28 14:56:52 2014 +0100

    multipointer: Make cursor updates device specific.
    
    Added framebuffer drawing of per-client cursors, made pixels2curs()
    work on 64-bit platforms as well.

:100644 100644 3115777956ab71133de7fa2c1fd7ea80b7e54d1a 8802dbf37201630d8105129bcfead250291765c2 M      cursor.c
:100644 100644 05986dd3cd8631e1c571fb86bc6f464d269d6a85 43c35c2c0d444254a42647de409dba324240e4bc M      cursor.h
:100644 100644 95c0444dad59540ab5a258e96eb11f6667c46423 7c327670f4967bdf2147da14aeb3aa24ada4f454 M      pointer.c
:100644 100644 77dd13a660f6f6211d09fd55a6bae2878cb9760d 68bbcfcd5e337a21a2f6091ed2cb68f7979dac99 M      screen.c
:100644 100644 aa23915669135ade608f6c5bc8d3a1db42fdece2 66a71c696ff206df57c962f8a8f713ea31a4470e M      xi2_devices.c
```

**Comment by bk138** [comment]
@djfd @apocalyptech can you pls provide 'uname -a' output on your machines?

**Comment by apocalyptech** [comment]
Sure thing; first the box that's actually running x11vnc:

Linux arrakis 4.10.3-1-ARCH #1 SMP PREEMPT Wed Mar 15 09:17:17 CET 2017 x86_64 GNU/Linux

And the box running tigervnc's vncviewer to access:

Linux dazhbog 4.10.3-1-ARCH #1 SMP PREEMPT Wed Mar 15 09:17:17 CET 2017 x86_64 GNU/Linux

**Comment by djfd** [comment]
x11vnc: ```Linux my-arch 4.10.6-1-ARCH #1 SMP PREEMPT Mon Mar 27 08:28:22 CEST 2017 x86_64 GNU/Linux```
remmina: ```Linux my-ubu 4.4.0-45-generic #66-Ubuntu SMP Wed Oct 19 14:12:37 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux```

**Comment by bk138** [comment]
Should be fixed by #49
